### PR TITLE
NUI-5753: standarize-and-document-how-we-mark-deprecations

### PR DESCRIPTION
**Pull request by madelyneh**

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/deprecation_report.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
```
Comment (Blakesalot) [inline .github/ISSUE_TEMPLATE/deprecation_report.md:1]: Do you mind renaming the file `breaking_change_task.md`

File: docs/DEPRECATIONS_AND_BREAKING_CHANGES.md [hunk 2]
```diff
@@ -9,3 +9,16 @@ to ensure the smoothest possible transition between major Nova versions.
 1. API breaking changes can be expected in conjunction with a Nova major version increase. Read the release notes carefully to look for the deprecated items that are set for removal. See the [CHANGELOG.md](./CHANGELOG.md) for details;
 2. Everything that has had a `@deprecated` decorator for a full major version will likely to be removed with the next major version increase. This will be also reflected in a migration guide.
 3. If a deprecated entity has been in this status for less than a full major version, it will be carried over into the next major version, and it will be removed in the one after it.
+
+## Our Deprecation Guidelines
+
+1. Create a new [ticket in github](https://github.com/solarwinds/nova/issues/new?assignees=&labels=&template=deprecation_report.md&title=) and fill in all the required information.
```
Comment (Blakesalot) [inline docs/DEPRECATIONS_AND_BREAKING_CHANGES.md:15]: Suggested tweak:

```suggestion
1. Create a new [breaking changed](https://github.com/solarwinds/nova/issues/new?assignees=&labels=&template=breaking_change_task.md&title=) ticket in github and fill in all the required information.
```

File: docs/DEPRECATIONS_AND_BREAKING_CHANGES.md [hunk 3]
```diff
@@ -9,3 +9,16 @@ to ensure the smoothest possible transition between major Nova versions.
...
+2. Inside a jsdoc style comment above the depreacted item
```
Comment (Blakesalot) [inline docs/DEPRECATIONS_AND_BREAKING_CHANGES.md:16]:
```suggestion
2. Inside a jsdoc style comment above the deprecated item
```

File: docs/DEPRECATIONS_AND_BREAKING_CHANGES.md [hunk 4]
```diff
@@ -9,3 +9,16 @@ to ensure the smoothest possible transition between major Nova versions.
...
+    - use <code>@deprecated</code>
```
Comment (Blakesalot) [inline docs/DEPRECATIONS_AND_BREAKING_CHANGES.md:17]:
```suggestion
    - use the <code>@deprecated</code> tag
```

File: docs/DEPRECATIONS_AND_BREAKING_CHANGES.md [hunk 5]
```diff
@@ -9,3 +9,16 @@ to ensure the smoothest possible transition between major Nova versions.
...
+    - state the version in which the deprecation occurred
```
Comment (Blakesalot) [inline docs/DEPRECATIONS_AND_BREAKING_CHANGES.md:18]:
```suggestion
    - state the major version in which the deprecation occurred
```

File: docs/DEPRECATIONS_AND_BREAKING_CHANGES.md [hunk 6]
```diff
@@ -9,3 +9,16 @@ to ensure the smoothest possible transition between major Nova versions.
...
+    - Include the recommended non-deprecated replacement
```
Comment (Blakesalot) [inline docs/DEPRECATIONS_AND_BREAKING_CHANGES.md:19]:
```suggestion
    - Include the recommended non-deprecated replacement that consumers can use instead
```

File: docs/DEPRECATIONS_AND_BREAKING_CHANGES.md [hunk 7]
```diff
@@ -9,3 +9,16 @@ to ensure the smoothest possible transition between major Nova versions.
...
+    - and the github id for the ticket you just created.
```
Comment (Blakesalot) [inline docs/DEPRECATIONS_AND_BREAKING_CHANGES.md:20]:
```suggestion
    - Include the github id for the ticket you just created
```

File: docs/DEPRECATIONS_AND_BREAKING_CHANGES.md [hunk 8]
```diff
@@ -9,3 +9,17 @@ to ensure the smoothest possible transition between major Nova versions.
 1. API breaking changes can be expected in conjunction with a Nova major version increase. Read the release notes carefully to look for the deprecated items that are set for removal. See the [CHANGELOG.md](./CHANGELOG.md) for details;
 2. Everything that has had a `@deprecated` decorator for a full major version will likely to be removed with the next major version increase. This will be also reflected in a migration guide.
 3. If a deprecated entity has been in this status for less than a full major version, it will be carried over into the next major version, and it will be removed in the one after it.
+
+## Our Deprecation Guidelines
```
Comment (Blakesalot) [inline docs/DEPRECATIONS_AND_BREAKING_CHANGES.md:13]:
```suggestion
## Steps for Introducing a Deprecation
```

**Review by baggiosw — APPROVED** [review]

**Review by Blakesalot — APPROVED** [review]

**Pull request merged**